src/components/CodeInput/index.js

```diff
@@ -0,0 +1,141 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { useUniqueIdentifier } from '../../libs/hooks';
+import useReduxForm from './../../libs/hooks/useReduxForm';
+import getNormalizedValue from './helpers/getNormalizedValue';
+import HelpText from '../Input/styled/helpText';
+import InputItems from './inputItems';
+import Label from '../Input/label';
+import RenderIf from '../RenderIf';
+import StyledFieldset from './styled';
+import StyledTextError from '../Input/styled/errorText';
+
+/**
+ * The CodeInput is an element that allows to fill a list of numbers, suitable for code validations.
+ * @category Form
+ */
+const CodeInput = React.forwardRef((props, ref) => {
+    const {
+        id,
+        name,
+        value,
+        label,
+        bottomHelpText,
+        length,
+        disabled,
+        required,
+        readOnly,
+        error,
+        tabIndex,
+        onClick,
+        onChange,
+        onFocus,
+        onBlur,
+        onKeyDown,
+        className,
+        style,
+    } = useReduxForm(props);
+
+    const inputId = id || useUniqueIdentifier('code-input');
+    const fieldsetName = name || useUniqueIdentifier('code-input-fieldset');
+
+    const handleOnChange = (inputValue, index, onChangeCallback) => {
+        onChange(getNormalizedValue(inputValue, index, value, length, onChangeCallback));
+    };
+
+    return (
+        <StyledFieldset className={className} name={fieldsetName} style={style} id={inputId}>
+            <Label
+                label={label}
+                hideLabel={!label}
+                required={required}
+                readOnly={readOnly}
+                inputId={inputId}
+            />
```

Comment:
You have to check the inputId is putting it to the fieldset, you should put it to the first input and fieldset id is the id of the props.